nav2_costmap_2d/test/integration/costmap_tester.cpp

```diff
@@ -52,13 +55,17 @@ class CostmapTester : public testing::Test {
     virtual void TestBody(){}
 
   private:
-    nav2_costmap_2d::Costmap2DROS costmap_ros_;
+    std::shared_ptr<nav2_costmap_2d::Costmap2DROS> costmap_ros_;
 };
 
-CostmapTester::CostmapTester(tf2_ros::Buffer& tf): costmap_ros_("test_costmap", tf){}
+CostmapTester::CostmapTester(tf2_ros::Buffer& tf)
+{
+  costmap_ros_ = std::make_shared<nav2_costmap_2d::Costmap2DROS>("test_costmap", tf);
+  //rclcpp::spin(costmap_ros_);
```

Comment:
Definitely shouldn't spin in the constructor. Should just delete the commented-out line.